eZ/Publish/Core/Repository/URLService.php

```diff
@@ -63,10 +67,15 @@ public function findUrls(URLQuery $query)
 
         $results = $this->urlHandler->find($query);
 
-        $items = [];
-        foreach ($results['items'] as $url) {
-            $items[] = $this->buildDomainObject($url);
-        }
+        $items = $results['items'];
+
+        $items = array_map(function ($url) {
+            return $this->buildDomainObject($url);
+        }, $items);
+
+        $items = array_values(array_filter($items, function ($url) {
+            return $this->permissionResolver->canUser('url', 'view', $url);
```

Comment:
After giving it more thought, despite internal discussion, it feels to me that checking this for each item separately instead is a performance issue.
The result of that check is always deterministic and independent of `$url`. The issue is role assignment limitation (Section, Subtree) which will return permission set instead of `true`.

@adamwojs @mikadamczyk wouldn't you agree that user is not authorized to `url/view` only if `hasAccess` returns `false`?

Reply:
We decided to change this to `        if ($this->repository->hasAccess('url', 'view') === false) {
            throw new UnauthorizedException('url', 'view');
        }`